runtime-decompiler/src/main/java/org/jrd/frontend/frame/main/MainFrameView.java

```diff
@@ -134,6 +143,43 @@ public String toString() {
         }
     }
 
+    private void setImageIcon() {
+        class BufferedImageTranscoder extends ImageTranscoder {
+            BufferedImage result;
+
+            @Override
+            public BufferedImage createImage(int width, int height) {
+                return new BufferedImage(width, height, BufferedImage.TYPE_INT_ARGB);
+            }
+
+            @Override
+            public void writeImage(BufferedImage image, TranscoderOutput to) throws TranscoderException {
+                this.result = image;
+            }
+        }
+
+        var transcoder = new BufferedImageTranscoder();
+
+        try (var file = getClass().getResourceAsStream("/icons/main-icon.svg")) {
+            TranscoderInput input = new TranscoderInput(file);
+            try {
+                transcoder.transcode(input, null);
+                Image originalImage = transcoder.result;
+                List<Image> images = new ArrayList<>();
+                images.add(originalImage);
+                for (int resolution = 32; resolution < originalImage.getHeight(null); resolution *= 2) {
+                    images.add(originalImage.getScaledInstance(resolution, resolution, Image.SCALE_SMOOTH));
+                }
+                mainFrame.setIconImages(images);
+
+            } catch (TranscoderException ex) {
+                throw new RuntimeException(ex);
+            }
+        } catch (IOException ex) {
+            throw new RuntimeException(ex);
+        }
```

Comment:
Do we really want to fail to start the app here because of the missing icon? I would opt for logging the problem and continuing with the default icon.